src/device-registry/services/nearest-device.js

```diff
@@ -0,0 +1,60 @@
+const { logElement } = require("../utils/log");
+
+const nearestDevices = {
+
+    findNearestDevices: ( devices, radius, latitude, longitude ) => {
+        
+        try {
+
+            let nearest_devices = []
+
+            devices.forEach(device => {
+
+                if ('latitude' in device && 'longitude' in device && device['isActive'] == true) {
+                    distance = nearestDevices.calculateDistance(latitude, longitude, device['latitude'], device['longitude']);
+
+                    if (distance < radius) {
+                        logElement("Found a node ", distance);
+                        nearest_devices.push(device);
+                    }
+                }
+          
+            });
+
+            return nearest_devices;
+        } catch (error) {
+            logElement("error", error);
+        }
+    },
+
+    calculateDistance: ( latitude1, longitude1, latitude2, longitude2) => {
+        
+        // distance between latitudes and longitudes
+        var latitudeDisatnce = nearestDevices.degreesToRadians(latitude2 - latitude1);
```

Comment:
Please either use `let` and `const` as declarations for variables, `var` is globally scoped which can be problematic since they can redeclared within their scope.